src/argilla/client/feedback/schemas/metadata.py

```diff
@@ -318,6 +347,23 @@ class FloatMetadataProperty(_NumericMetadataPropertySchema):
     min: Optional[float] = None
     max: Optional[float] = None
 
+    def _check_allowed_value_type(self, value: Any) -> Any:
+        if value is not None:
+            if isinstance(value, (int, float)):
+                return value
+            raise TypeError(
+                f"Provided '{self.name}={value}' of type {type(value)} is not valid, "
+                "only values of type `int` or `float` are allowed."
+            )
+        return value
```

Comment:
Another similar suggestion:

```suggestion
    def _check_allowed_value_type(self, value: Any) -> Any:
        if value is None or isinstance(value, (int, float)):
            return value
            
        raise TypeError(
            f"Provided '{self.name}={value}' of type {type(value)} is not valid, "
            "only values of type `int` or `float` are allowed."
        )
```